src/systemd_ctypes/libsystemd.py

```diff
@@ -86,18 +86,44 @@ class id128(Structure):
     (staticmethod, negative_errno, 'default', [POINTER(sd.event_p)]),
 ])
 
+def signature_is_valid(str):
+    try:
+        parse_signature(str)
```

Comment:
`parse_signature()` only produces correct results for correct signatures and was never designed to catch errors.  It'll happily parse non-sense like `a{vs}` or `a{sss}` or `ZZZ` or even `!@#$`.